src/Datasource/Exception/RuleAlreadyExistsException.php

```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Datasource\Exception;
+
+use Cake\Core\Exception\CakeException;
+
+/**
+ * Exception raised when a rule with the same name already exists
+ */
+class RuleAlreadyExistsException extends CakeException
```

Comment:
Given how uncommon this issue is and it's very likely to be caught at development time itself, I am not sure a separate exception is needed. We could just throw `CakeException`.

Let's see what others have to see.

Reply:
I agree with @ADmad. This isn't a type that will have application level catch blocks and we won't have a specific error page either. `CakeException` should be good enough here.